app/modules/fileuploads/models.py

```diff
@@ -0,0 +1,174 @@
+        filepath = self.get_absolute_path()
+        with db.session.begin():
+            db.session.delete(self)
+        if os.path.exists(filepath):
+            os.remove(filepath)
+
+    # this is singular, so single (tus)path required
+    #   note: this is 'path' from { transaction_id, path } in tus args.  sorry so many things called path.
+    @classmethod
+    def create_fileupload_from_tus(cls, transaction_id, owner, path):
+        assert transaction_id is not None
+        assert path is not None
+        source_path = _tus_filepath_from(transaction_id, path)
+        fup = FileUpload.create_fileupload_from_path(source_path, owner)
+        _tus_purge(transaction_id)
+        return fup
+
+    # plural paths is optional (will do all files in dir if skipped)
+    @classmethod
+    def create_fileuploads_from_tus(cls, transaction_id, owner, paths=None):
+        assert transaction_id is not None
+        source_paths = _tus_filepaths_from(transaction_id, paths)
+        if source_paths is None or len(source_paths) < 1:
+            return None
+        fups = []
+        for source_path in source_paths:
+            fups.append(FileUpload.create_fileupload_from_path(source_path, owner))
+        _tus_purge(transaction_id)
+        return fups
+
+    @classmethod
+    # default behavior is to *move*
+    def create_fileupload_from_path(cls, source_path, owner, copy=False):
+        assert source_path is not None
+        fup = FileUpload()
+        fup.owner = owner
+        if copy:
+            fup.copy_from_path(source_path)
+        else:
+            fup.move_from_path(source_path)
+        return fup
+
+    def copy_from_path(self, source_path):
+        assert os.path.getsize(source_path) > 0
+        log.debug(
+            'copy_from_path: %r to %r for %r'
+            % (
+                source_path,
+                self.get_absolute_path(),
+                self,
+            )
+        )
+        os.makedirs(self.dirname(), exist_ok=True)
+        shutil.copyfile(source_path, self.get_absolute_path())
+        self.derive_mime_type()
+
+    def move_from_path(self, source_path):
+        assert os.path.getsize(source_path) > 0
+        log.debug(
+            'move_from_path: %r to %r for %r'
+            % (
+                source_path,
+                self.get_absolute_path(),
+                self,
+            )
+        )
+        os.makedirs(os.path.dirname(self.get_absolute_path()), exist_ok=True)
+        shutil.move(source_path, self.get_absolute_path())
+        self.derive_mime_type()
+
+    # note: this may not exist (we may just need it as a target for copy/move)
+    def get_absolute_path(self):
+        base_path = current_app.config.get('FILEUPLOAD_BASE_PATH', None)
+        assert base_path is not None
+        assert self.guid is not None
+        user_path = '00000000-0000-0000-0000-000000000000'  # for owner-less
+        if self.owner is not None:
+            user_path = str(self.owner.guid)
```

Comment:
By the way is it important to include the owner guid in the path or even in the object?  Are we going to have fileupload objects that are only linked to a user but not to anything else?

Reply:
no, i would not call this important specifically.  it was just an easy way to kind of partition things a little -- also add a bit of tracking?

Reply:
.owner now... gone!